### ISSUE TITLE
Official recommendations for FW migration

### ISSUE THREAD

**Issue by tcharp38** [fairecasoimeme/ZiGate#356]
Not an issue but a clarification.

What are your recommendations to migrate from an older FW to 31e ?
Is it supposed to be transparent ?

3.1d has just been a nightmare for lot's of people including me. Forced to reset PDM et restart from scratch.

Moreover what about reversing back to previous version if too many troubles with 3.1e ?
I assume that it is NOT recommended to switch back to older version unless resetting PDM (and therefore reincluding all EQ).

Please clarify official recommendations.
Thanks

**Comment by pipiche38** [comment]
I'll leave @fairecasoimeme  to comment, but here is my understanding

you should not have any issue to upgrade from 31a to 31e, and of course from 31d to 31e

Going back and forth between 31e <-> 31d should also works as there is no change in the PDM

Only the move to ZiGate v3.1e (Optimised PDM) would required Erase EEPROM and Erase PDM and no fall back to 31e would be possible.

**Comment by schrodingersket** [comment]
What exactly does the Optimized PDM do? It looks like both versions of `3.1e` are built from the same commit; is there a different `.zpsconfig` file that the Optimized PDM variation is built from?

**Comment by schrodingersket** [comment]
Got it. Will there now be two versions for each release moving forward (one for the unoptimized PDM structure and one for the optimized)?On May 7, 2021 9:39 AM, Pipiche ***@***.***> wrote:
indeed both are based on the same 31e code and so functionality, but the OptyPDM has a differnt set of table size in order to follow NXP sizing recommendation. We have also review the APS, nPDU and aPDU size

**Comment by pipiche38** [comment]
yes, that is correct, you should expect always a OptiPDM and a Legacy version of the Zigate V1 firmware. As the ZiGate V2 it is anyhow a different firmware and it is based on a different PDM
